### DISCUSSION TITLE
Theoretical improvement.

### DISCUSSION THREAD

**Discussion by commander-trashdin** [adlai/chanl#9] [Ideas]
Hello, I'm using this as an advised way to contact you. Do you think Chanl could (in terms of performance) benefit from statically dispatched function system (instead of generics) and compile time generated structs with precise types instead of classes?
Thanks in advance.

**Comment by adlai** [comment]
Apologies for the delayed reply... as you may have noticed, I took some time away from active maintaining of this project.

> Do you think Chanl could (in terms of performance) benefit from statically dispatched function system (instead of generics) and compile time generated structs with precise types instead of classes?

Yes! Not only would that improve performance, it also _could_ reduce the surface area of undefined behavior, provided that the replacement for the current CLOS-based implementation would be thoroughly tested... the bugs described in zkat/chanl#11 still do exist.

I don't think you and I have communicated before, so I'm not familiar with your background in CL programming; however, I do see from your recent GitHub activity that you are interested in long-term CL work. Are you interesting in collaborating to improve this project?

**Comment by commander-trashdin** [comment]
Wow. I already gave up on recieving a reply to this and even completely forgot about this thread.
Anyway, since you are here... 
Here are libraries I and a couple of other people are working on:
https://github.com/digikar99/polymorphic-functions  <--- this is the main thing, the aforementioned "statically dispatched function system".
Here is my project that is completely based on it:
https://github.com/lisp-polymorph
Basically trying to rewrite standard library in terms of polymorphic functions. Also add some new stuff that I think it is lacking. 
Is there a better way to contact you, so we could discuss this in depth?

**Comment by adlai** [comment]
> Is there a better way to contact you, so we could discuss this in depth?

Discussing things of direct influence to the project in public has the advantage of forming publicly readable conversations... because there are several email addresses listed in various git metadata, I'll write to the one listed in your GitHub profile and hope it won't get eaten by Google's amazing spam filter.

**Comment by adlai** [comment]
re: https://github.com/mdbergmann/cl-gserver/issues/94#issuecomment-2402205666

possible names evoking the appropriate violation of any unwritten static prejudices...

- destruct/oy
- halt, break, terminate, obvious
- quvderat-demonstrandum
- :keyword
- finalise-now

**Comment by adlai** [comment]
re: adlai/scalpl#40

OOM crashes occur frequently enough to place one of my contracts in jeopardy, so reducing the footprint of dependencies -- most notably, ChanL -- has returned to higher priority.
